src/features/filter.ts

```diff
@@ -1,10 +1,52 @@
-export const actions = { /* put action creators here */};
+import { Todo } from '../types/Todo';
 
-const filterReducer = () => {
-  return {
+export enum Filter {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+}
+
+type FilterType = { type: 'set/FILTER'; filter: Filter };
+type QueryType = { type: 'set/QUERY'; query: string };
+type FilterTodos = { type: 'set/TODOS', todos: Todo[] };
+
+const setFilter = (filter:Filter)
+: FilterType => ({ type: 'set/FILTER', filter });
+const setQuery = (query:string)
+: QueryType => ({ type: 'set/QUERY', query });
+const setFilterTodos = (todos: Todo[])
+: FilterTodos => ({ type: 'set/TODOS', todos });
+
+export const actions = { setFilter, setQuery, setFilterTodos };
+
+type State = {
+  query: string,
+  filter: Filter,
+  todos: Todo[],
+};
+type Actions = FilterType | QueryType | FilterTodos;
+
+export const filterReducer = (
+  state: State = {
     query: '',
-    status: 'all',
-  };
+    filter: Filter.ALL,
+    todos: [],
+  },
+  action: Actions,
+) => {
+  switch (action.type) {
+    case 'set/QUERY':
+      return { ...state, query: action.query };
+
+    case 'set/FILTER':
+      return { ...state, filter: action.filter };
+
+    case 'set/TODOS':
+      return { ...state, todos: action.todos };
```

Comment:
shouldn't be here - just remove it at all

Reply:
why should I remove 'set/TODOS', if I have a filtered array stored in the reducer

Reply:
Or should I store the filtered array in state?